src/tests/integration/kcfg/test_imp.py

```diff
@@ -71,6 +71,91 @@
     ),
 )
 
+IMPLICATION_FAILURE_TEST_DATA: Final = (
+    (
+        'different-cell',
+        ('int $n ; $n = 0 ;', '.Map'),
+        ('int $n ; $n = 1 ;', '.Map'),
+        (
+            'Structural matching failed, the following cells failed individually (abstract => concrete):\n'
+            'K_CELL: ( int $n , .Ids ; $n = 1 ; => int $n , .Ids ; $n = 0 ; )'
+        ),
+    ),
+    (
+        'different-cell-2',
+        ('int $n ; $n = X:Int ;', '.Map'),
+        ('int $n ; $n = 1 ;', '.Map'),
+        (
+            'Structural matching failed, the following cells failed individually (abstract => concrete):\n'
+            'K_CELL: ( int $n , .Ids ; $n = X:Int ; => int $n , .Ids ; $n = 1 ; )'
```

Comment:
I guess this needs to be flipped.

Can we also just change it everywhere to say `abstract => consequent` and `concrete => antecedent`? It's confusing to have different terminology for them, should just use the `antecedent #Implies consequent` everwhere.

When outputting this error message, we should do it in implication order, with an actual implication. So we ideally would output:

```suggestion
            'K_CELL: ( int $n , .Ids ; $n = 1 ; #Implies int $n , .Ids ; $n = X:Int ; )'
```

This will require, after doing `push_down_rewrites` and `no_cell_rewrite_to_dots`, to make a routine called `replace_rewrites_with_implies`, and run that on the term. It can look like this:

```
def replace_rewrites_with_implies(kast: KInner) -> KInner:
    def _replace_rewrites_with_implies(_kast: KInner) -> KInner:
        if type(_kast) is KRewrite:
            return mlImplies(_kast.lhs, _kast.rhs)
        return _kast
    return bottom_up(_replace_rewrites_with_implies, kast)
```